### PR TITLE
A0-1718 remove old authentications

### PR DESCRIPTION
**Pull request by krzysztofziobro**
# Description
Removing old authentications.

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: finality-aleph/src/network/session/compatibility.rs [hunk 1]
```diff
@@ -19,129 +16,36 @@ type ByteCount = u16;
...
-            NewOnly(authentication) => vec![V2(authentication)],
-            LegacyOnly(legacy_authentication) => {
-                vec![V1(AuthenticationBroadcast(legacy_authentication))]
-            }
-            Both(authentication, legacy_authentication) => vec![
-                V2(authentication),
-                V1(AuthenticationBroadcast(legacy_authentication)),
-            ],
-        }
+        vec![V2(authentication)]
```
Comment (timorl) [inline finality-aleph/src/network/session/compatibility.rs:29]: The `use` above is a bit of an overkill, just spell the whole thing out in one line. ;p

File: finality-aleph/src/network/session/compatibility.rs [hunk 2]
```diff
@@ -19,129 +16,36 @@ type ByteCount = u16;
...
 
-/// One of the possible messages we could have gotten as part of discovery.
-/// Ignores whether the old authentication was a broadcast or not, we don't send the
-/// non-broadcasts anymore anyway, and treating them as broadcasts doesn't break anything.
-#[derive(Clone, Debug, PartialEq, Eq, Hash)]
-pub enum DiscoveryMessage<M: Data, A: AddressingInformation + TryFrom<Vec<M>> + Into<Vec<M>>> {
-    Authentication(Authentication<A>),
-    LegacyAuthentication(LegacyAuthentication<M>),
-}
+pub type DiscoveryMessage<A> = Authentication<A>;
```
Comment (timorl) [inline finality-aleph/src/network/session/compatibility.rs:33]: Ooh, nice hack, love it.

File: finality-aleph/src/network/session/manager.rs [hunk 3]
```diff
@@ -7,13 +7,14 @@ use std::{
 use futures::channel::mpsc;
 use log::{debug, info};
 
+use super::Authentication;
```
Comment (timorl) [inline finality-aleph/src/network/session/manager.rs:10]: Ah, we generally avoid `super` (not sure whether that's actually good, but this is the current convention). In particular some stuff below is imported from `super`, but without using the keyword, so this is now inconsistent even within this one file.

**Review by timorl — APPROVED** [review]
Nice, thanks for the trimming work. <3

File: finality-aleph/src/network/session/compatibility.rs (hunk 1 above)
Comment (krzysztofziobro) [reply finality-aleph/src/network/session/compatibility.rs]: fixed

File: finality-aleph/src/network/session/manager.rs (hunk 3 above)
Comment (krzysztofziobro) [reply finality-aleph/src/network/session/manager.rs]: fixed

File: finality-aleph/src/network/session/compatibility.rs (hunk 2 above)
Comment (krzysztofziobro) [reply finality-aleph/src/network/session/compatibility.rs:32]: thx

**Review by maciejnems — APPROVED** [review]
Very Nice

**Pull request merged**